app/models/page_view.rb

```diff
@@ -2,11 +2,11 @@ class PageView < ApplicationRecord
   include AlgoliaSearch
 
   belongs_to :user, optional: true
-  belongs_to :article
+  belongs_to :article, optional: true
```

Comment:
Could we expand on this constraint to say that if an `article_id` is not present then the page must have a path, url, and category? This was preventing us from making a lot of invalid page views before and I worry with it removed we will open ourselves up for recording those "invalid" page views.

Reply:
That makess sense